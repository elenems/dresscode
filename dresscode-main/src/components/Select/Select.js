import ReactSelect from "react-select";
import React from "react";
import PropTypes from "prop-types";

function Select(props) {
  return (
    <div className="select">
      <span className="label-style">{props.label}</span>
      <ReactSelect
        value={props.value}
        onChange={props.handleChange}
        options={props.options}
      />
    </div>
  );
}

Select.propTypes = {
  handleChange: PropTypes.func.isRequired,
  options: PropTypes.array,
  label: PropTypes.string
};

export default React.memo(Select);
