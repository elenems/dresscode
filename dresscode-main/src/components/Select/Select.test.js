import React from "react";
import { shallow } from "enzyme";
import Select from "./Select";

describe("Select", () => {
  const props = {
    label: "condition",
    value: "new",
    handleChange: () => {},
    options: []
  };
  it("Matches snapshot", () => {
    const wrapper = shallow(<Select {...props} />).debug();
    expect(wrapper).toMatchSnapshot();
  });
});
