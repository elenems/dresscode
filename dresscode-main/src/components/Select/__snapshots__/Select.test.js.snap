// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select Matches snapshot 1`] = `
"<div className=\\"select\\">
  <span className=\\"label-style\\">
    condition
  </span>
  <StateManager value=\\"new\\" onChange={[Function: handleChange]} options={{...}} defaultInputValue=\\"\\" defaultMenuIsOpen={false} defaultValue={{...}} />
</div>"
`;
